synapse/rest/media/v1/_base.py

```diff
@@ -153,7 +163,8 @@ def _quote(x):
     # select private. don't bother setting Expires as all our
     # clients are smart enough to be happy with Cache-Control
     request.setHeader(b"Cache-Control", b"public,max-age=86400,s-maxage=86400")
-    request.setHeader(b"Content-Length", b"%d" % (file_size,))
+    if file_size is not None:
+        request.setHeader(b"Content-Length", b"%d" % (file_size,))
```

Comment:
This is due to `respond_with_responder` accept `None` which then gets passed through to here. I don't know if that *actually* gets called with `None` ever though. 😢